### PR TITLE
Afform - Misc tweaks, validation & bug fixes

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
This adds basic validation to `server_route` in the Afform GUI, only permitting paths beginning with "civicrm/".
It also re-labels the menu item and page title to "Form Builder"
And resets `server_route` and `is_dashlet` when cloning.
And fixes several bugs in Afform & Search Kit, including a couple regressions within the 5.35 cycle.

### CODE REVIEW

**Comment by seamuslee001** [comment]
@colemanw should we also have any validation to prevent multiple forms using the same exact path?

**Comment by colemanw** [comment]
@seamuslee001 yes I'll do that in a follow-up PR. That will be done server-side and this is client-side.

**Comment by seamuslee001** [comment]
@colemanw I tested this and I struggled to see any obvious errors when I put in a url that isn't `civicrm/*` as per the attached screen grab

afform-url-validaton

**Comment by colemanw** [comment]
@seamuslee001 yea the validation is "working" in that it prevents the invalid value from being saved (as you see on the list view after saving)... but something is preventing html5 validation msgs from appearing. I'm trying to figure that out...

**Comment by seamuslee001** [comment]
@totten @colemanw I tested the latest version and can confirm that the html validation is working correctly now, code changes look good to me

File: ext/afform/admin/ang/afGuiEditor/config-form.html [hunk 1]
```diff
@@ -18,10 +18,10 @@
 
 <div class="form-group">
   <label for="af_config_form_server_route">
-    {{:: ts('Path:') }}
+    {{:: ts('Page:') }}
   </label>
-  <input ng-model="afform.server_route" class="form-control" id="af_config_form_server_route" />
-  <p class="help-block">{{:: ts('Expose the form as a standalone page on the web site. (Example: "civicrm/my-form")') }}</p>
+  <input ng-model="afform.server_route" class="form-control" id="af_config_form_server_route" pattern="^civicrm\/[-0-9a-zA-z\/_]+$" onfocus="this.value = this.value || 'civicrm/'" onblur="if (this.value === 'civicrm/') this.value = ''" title="{{:: ts('Path must begin with &quot;civicrm/&quot;') }}">
```
Comment (totten) [inline ext/afform/admin/ang/afGuiEditor/config-form.html]: I think the `pattern` constraint prevents any data from getting out of the model (huzzah). But to get the output styled, I think the key things are from BS3 and AngularJS:

* Bootstrap3 (https://getbootstrap.com/docs/3.4/css/)
    * You should put `has-error` on any `form-group` with validation errors
* AngularJS Validation (https://docs.angularjs.org/guide/forms#custom-validation): 
    * Enforces `pattern` attribute
    * The relevant `[ng-model]` element will automatically get the class `ng-valid` or `ng-invalid`.
    * You can inspect validation status via `FORM_NAME.INPUT_NAME.$error.TYPE` (e.g. `config_form.server_route.$error.pattern`). This is most useful for conditional messaging
        ```html
          <p class="help-block" ng-show="config_form.server_route.$error.pattern">ZOMG</p>
        ```

I pushed up a patch which does a quick fix for this element. So you can see reasonable styling for the error:

Screen Shot 2021-02-03 at 1 16 48 AM

Of course, one would prefer to have a generic rule (erm, `directive`) which says, "Within a Bootstrap3-based screen, watch for `.ng-invalid` and toggle `.has-error` higher-up." There are some examples along those lines in https://stackoverflow.com/questions/14348384/reconcile-angular-js-and-bootstrap-form-validation-styling (*I didn't put any of those example snippets the PR -- something felt a little off, so one should do it more thoughtfully, but they give the gist.*)

**Comment by totten** [comment]
> (@seamuslee001) I tested the latest version and can confirm that the html validation is working correctly now, code changes look good to me

Cool :+1:

I'm inclined to merge. Validation is now functional for Seamus+me.  FWIW, I spot-checked some of the other commits (a little random `r-run` and `r-code`); I think Seamus did similarly; and neither spotted anything else. Coleman wanted to get the fixes in before freeze. And this code only affects the client-side of beta-stage extensions, so I don't think we need to be super-rigorous.

There's an opportunity to handle the `ng-invalid`/`has-error` stuff more generally, but I don't think it's worth blocking the other stuff in here.

Merge-on-pass.

**Pull request merged**